frontend/src/components/Searchbar/Searchbar.tsx

```diff
@@ -17,104 +17,157 @@ import IconButton from '../IconButton'
 
 export interface SearchbarProps extends Omit<InputProps, 'onChange'> {
   /**
-   * Function to be invoked when user presses enter (to search).
+   * Function to be invoked when user presses enter (to search) or clicks the
+   * 'search' icon.
    * @param searchValue value of the search input
    */
   onSearch: (searchValue: string) => void
 
   /**
-   * Whether the searchbar is expanded or not.
-   * @note This should be `true` if the `onSearchIconClick` function prop is not
-   * provided, or the searchbar will not be usable.
+   * Whether the searchbar is expandable and collapsable.
+   * If this is `false`, the searchbar will be permanently expanded,
+   * regardless of the value passed for isExpanded.
+   * @defaultValue `true`
    */
-  isExpanded: boolean
+  isExpandable?: boolean
 
   /**
-   * Optional. Function to be invoked when the search icon is clicked.
-   * If provided, the search icon will be clickable.
+   * Whether the searchbar is initially expanded or not.
+   * This value will be overridden and set to `true` if `isExpandable` is
+   * `false` (otherwise, the searchbar will not be usable).
+   * @defaultValue `false`
    */
-  onSearchIconClick?: () => void
+  isExpanded?: boolean
 
-  defaultValue?: string
-  onChange?: (newValue: string) => void
+  /**
+   * Optional. Function to be invoked when the 'expand' search icon is clicked.
+   * Ignored if `isExpandable` is set to `false`.
+   */
+  onExpandIconClick?: () => void
+
+  /**
+   * Optional. Function to be invoked when the 'collapse' X icon is clicked.
+   * Ignored if `isExpandable` is set to `false`. Note that this clears any
+   * value from the searchbar input but *does not* call `onValueChange`.
+   */
+  onCollapseIconClick?: () => void
+
+  /**
+   * Initial value in the searchbar input.
+   */
   value?: string
 
-  rightElement?: React.ReactElement
+  /**
+   * Function to be invoked when the value in the searchbar input changes (but
+   * the search button has not been clicked).
+   */
+  onValueChange?: (newValue: string) => void
```

Comment:
this is fairly confusing, since there is already the `onChange` prop. What is the difference between `onValueChange` and `onChange`?

Reply:
there's no more `onChange` i believe. I renamed it because `onChange` feels too uninformative as a prop name - it could be a change to whether its expanded, or what the search term was when the user pressed enter

Reply:
opinion: `onChange` is a standard input prop though, and is what developers using inputs would expect. I think we should stick with `onChange`, especially when internally we are assigning onValueChange to yet another internal `onChange` prop

Reply:
hmm, i think that's reasonable, can just keep the documentation for reference